tock/hours/views.py

```diff
@@ -617,43 +616,40 @@ def get_object(self, queryset=None):
         return obj
 
     def clean_task_data(self, float_people_id):
-        # Get Float task data for week corresponding w/ ReportingPeriod
-        # instance.
+        # First day of reporting period.
+        start_date = datetime.datetime.strptime(
+            self.kwargs['reporting_period'], "%Y-%m-%d"
+        ).date()
+        # Last day of reporting period.
+        end_date = start_date + datetime.timedelta(days=6)
+        # Get Float task data for week corresponding start and end dates.
+        # Assumes a reporting period is 1 week.
         r = get_float_data(
             endpoint='tasks',
             params={
-                'weeks': base.FLOAT_API_TASK_WEEKS,
-                'start_day': datetime.datetime.strptime(
-                    self.kwargs['reporting_period'], "%Y-%m-%d"
-                ).date()
+                'weeks': 1,
+                'start_day': start_date
             }
         )
-        if r.status_code != 200:
+        if not r:
             return None
-        else:
-            task_data = r.json()
-        #From response, derive meta data and tasks for the period.
-        f_start_date = datetime.date(
-            int(task_data['start_yr']), 1, 1) + datetime.timedelta(
-            days=int(task_data['start_doy']
-            )-1)
-        f_end_date = f_start_date + datetime.timedelta(
-            days=(base.FLOAT_API_TASK_WEEKS * 6))
+        task_data = r.json()['people']
         clean_data = {
-            'metadata': {
-                'float_period_start': f_start_date,
-                'float_period_end': f_end_date
-            },
-            'tasks': [ item for sublist in \
-                [ ii for ii in [i['tasks'] for i in task_data['people'] \
-                if i['people_id'] == float_people_id] ] \
-            for item in sublist ]
+            # Get all of the tasks (ii) associated with a Float user if the
+            # user's Float people_id matches their Tock float_people_id.
+            # See https://github.com/floatschedule/api/blob/805663be98c9f48a275c9a13e824b0d6701df398/Sections/tasks.md.
+            'tasks': flatten(
+                [ ii for ii in [i['tasks'] for i in task_data \
+                    if int(i['people_id']) == float_people_id] ]
```

Comment:
Ah, so a Float user will always have a `people_id` (at least as far as I know...). The biggest hangup here is matching a Tock user's `User.username` with a Float `im` value (it was the only field that can be edited in Float via the GUI; Float admins manually enter a user's `User.username` into this field). _If_, however, Tock cannot match a `User.username` with a Float `im` value, an empty list is returned [here](https://github.com/18F/tock/pull/630/files/a1f789f9703054a0266e577a40ea5ae819437c0c..8843b820bf4ab1fa5fa98c32efa80950c1471d7f#diff-bbf773df48e042c9e0b2f421ddc3ee38R675), with nothing save to the user's UserData.float_people_id (thus avoiding the potential for a `ValueError: invalid literal for int()...`).

I know, a bit confusing.